rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -2866,29 +2522,16 @@ rmw_service_server_is_available(
   RMW_CHECK_ARGUMENT_FOR_NULL(client, RMW_RET_INVALID_ARGUMENT);
   RMW_CHECK_ARGUMENT_FOR_NULL(client->data, RMW_RET_INVALID_ARGUMENT);
   RMW_CHECK_ARGUMENT_FOR_NULL(is_available, RMW_RET_INVALID_ARGUMENT);
-
-  rmw_zenoh_cpp::rmw_client_data_t * client_data =
-    static_cast<rmw_zenoh_cpp::rmw_client_data_t *>(client->data);
+  rmw_zenoh_cpp::ClientData * client_data =
+    static_cast<rmw_zenoh_cpp::ClientData *>(client->data);
   if (client_data == nullptr) {
     RMW_SET_ERROR_MSG_WITH_FORMAT_STRING(
       "Unable to retreive client_data from client for service %s", client->service_name);
     return RMW_RET_INVALID_ARGUMENT;
   }
 
-  std::string service_type = client_data->request_type_support->get_name();
-  size_t suffix_substring_position = service_type.find("Request_");
```

Comment:
So this check was moved inside `ClientData::make()` but there was typo there anyways which I have fixed https://github.com/ros2/rmw_zenoh/pull/293/commits/4c9b1069629bfd2daea1f3e14ef5376dfca3f356. It would not have caused any problems since the substring we extract will be the same but now it's semantically correct.